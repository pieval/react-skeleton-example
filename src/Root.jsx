/* eslint "react/forbid-prop-types":0 */

import React, { PropTypes } from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import DevTools from './containers/Devtools';

const Root = (props) => {
  const { store, routes, history } = props;
  return (
    <Provider store={store}>
      <div>
        <Router history={history}>
          {routes()}
        </Router>
        { (process.env.NODE_ENV === 'development') ? <DevTools /> : null }
      </div>
    </Provider>
  );
};

Root.propTypes = {
  store: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  routes: PropTypes.func.isRequired,
};

export default Root;
