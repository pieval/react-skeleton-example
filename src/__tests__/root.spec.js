/* eslint no-undef: 0 */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import Root from '../Root';

describe('<Root />', () => {
  describe('Rendering', () => {
    const store = {
      name: 'TEST STORE',
      subscribe: () => {},
      dispatch: () => {},
      getState: () => {},
    };
    const history = {
      name: 'TEST HISTORY',
    };

    it('should render with default props / SNAPSHOT TEST', () => {
      const root = shallow(
        <Root
          store={store}
          history={history}
          routes={() => { 'TEST_ROUTES'; }}
        />);
      expect(shallowToJson(root)).toMatchSnapshot();
    });
    it('should has a router and call the routes method to get routes', () => {
      const routes = jest.fn();
      const root = shallow(
        <Root
          store={store}
          history={history}
          routes={routes}
        />);
      expect(root.find(Provider)).toHaveLength(1);
      expect(root.find(Router)).toHaveLength(1);
      expect(routes).toHaveBeenCalledTimes(1);
    });
  });
});
