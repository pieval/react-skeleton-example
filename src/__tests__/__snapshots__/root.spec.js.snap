// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> Rendering should render with default props / SNAPSHOT TEST 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "name": "TEST STORE",
      "subscribe": [Function],
    }
  }
>
  <div>
    <Router
      history={
        Object {
          "name": "TEST HISTORY",
        }
      }
      render={[Function]}
    />
  </div>
</Provider>
`;
